bentoml/_internal/bento/build_config.py

```diff
@@ -58,33 +59,74 @@ def _convert_python_version(py_version: t.Optional[str]) -> t.Optional[str]:
     if not isinstance(py_version, str):
         py_version = str(py_version)
 
-    match = re.match(r"^(\d+)\.(\d+)(?:\.\w+)?", py_version)
+    match = re.match(r"^(\d{1})\.(\d{,2})(?:\.\w+)?$", py_version)
     if match is None:
         raise InvalidArgument(
             f'Invalid build option: docker.python_version="{py_version}", python '
             f"version must follow standard python semver format, e.g. 3.7.10 ",
         )
     major, minor = match.groups()
-    return f"{major}.{minor}"
+    target_python_version = f"{major}.{minor}"
+    if target_python_version != py_version:
+        logger.warning(
+            "BentoML will install the latest python%s instead of the specified version %s. To use the exact python version, use a custom docker base image. See https://docs.bentoml.org/en/latest/concepts/bento.html#custom-base-image-advanced",
+            target_python_version,
+            py_version,
+        )
+    return target_python_version
 
 
 def _convert_cuda_version(
     cuda_version: t.Optional[t.Union[str, int]]
 ) -> t.Optional[str]:
-    if cuda_version is None or cuda_version == "":
+    if cuda_version is None or cuda_version == "" or cuda_version == "None":
```

Comment:
cuda_version will be str "None" if user accidentally has `cuda_version: null` in the bentofile.yaml

Reply:
maybe we can revisit this and have a better null handling @aarnphm